app/src/main/java/com/example/admin/personallibrarycatalogue/data/LibraryDatabaseHelper.java

```diff
@@ -0,0 +1,197 @@
+        super(context, DATABASE_NAME, null, DATABASE_VERSION);
+    }
+
+    /**
+     * Creates table to hold book data
+     */
+    @Override
+    public void onCreate(SQLiteDatabase sqLiteDatabase) {
+
+        final String SQL_CREATE_BOOKS_TABLE = "CREATE TABLE " + BooksTable.TABLE_NAME + " (" +
+                BooksTable._ID + " INTEGER PRIMARY KEY AUTOINCREMENT," +
+                BooksTable.AUTHOR + " TEXT NOT NULL, " +
+                BooksTable.TITLE + " TEXT NOT NULL, " +
+                BooksTable.DESCRIPTION + " TEXT, " +
+                BooksTable.COVER + " BLOB );";
+
+        sqLiteDatabase.execSQL(SQL_CREATE_BOOKS_TABLE);
+    }
+
+
+    @Override
+    public void onUpgrade(SQLiteDatabase sqLiteDatabase, int oldVersion, int newVersion) {
+        sqLiteDatabase.execSQL("DROP TABLE IF EXISTS " + DatabaseContract.BooksTable.TABLE_NAME);
+        onCreate(sqLiteDatabase);
+    }
+
+    public int getId(String title, String author) {
+        if ((title != null) && (author != null)) {
+
+            int id = -1;
+            SQLiteDatabase db = this.getWritableDatabase();
+
+            String selectQuery = "SELECT * FROM " + BooksTable.TABLE_NAME +
+                    " WHERE " + BooksTable.TITLE + " =  \"" + title + "\"" +
+                    " AND " + BooksTable.AUTHOR + " =  \"" + author + "\"";
+
+            Cursor cursor = db.rawQuery(selectQuery, null);
+
+            if (cursor.moveToFirst()) {
+                id = (cursor.getInt(cursor.getColumnIndex(BooksTable._ID)));
+            }
+
+            db.close();
+            return id;
+        } else {
+            throw new NullPointerException("Passed title or author is null");
+        }
+    }
+
+    public void deleteBook(int id) {
+        SQLiteDatabase db = this.getWritableDatabase();
+        db.delete(BooksTable.TABLE_NAME, BooksTable._ID + " = " + id, null);
+    }
+
+    public Book getBookById(int id) {
+        SQLiteDatabase db = this.getWritableDatabase();
+
+        Book book = new Book();
+
+        String selectQuery = "SELECT * FROM " + BooksTable.TABLE_NAME +
+                " WHERE " + BooksTable._ID + " = " + id;
+
+        Cursor cursor = db.rawQuery(selectQuery, null);
+
+        if (cursor.moveToFirst()) {
+            book.setTitle(cursor.getString(cursor.getColumnIndex(BooksTable.TITLE)));
+            book.setAuthor(cursor.getString(cursor.getColumnIndex(BooksTable.AUTHOR)));
+
+            if ((cursor.getString(cursor.getColumnIndex(BooksTable.DESCRIPTION))) != null) {
+                book.setDescription(cursor.getString(cursor.getColumnIndex(BooksTable.DESCRIPTION)));
+            }
+
+            if (cursor.getBlob(cursor.getColumnIndex(BooksTable.COVER)) != null) {
+                book.setCover(cursor.getBlob(cursor.getColumnIndex(BooksTable.COVER)));
+            }
+        }
+
+        db.close();
+        return book;
+    }
+
+    public Book getBook(String title, String author) {
```

Comment:
Guess to remove this method cause we can't rely on it. Or it should return the list of matched books instead.